### PR TITLE
Add permissions system

### PR DESCRIPTION
**Pull request by markbreedlove**
Add a permissions system, with administrator roles.

See https://issues.dp.la/issues/8163

The specification was to add a class of view-only users (who can view unpublished sets, but not modify or create them), and a permissions system is necessary to achieve this.

### CODE REVIEW

File: app/views/registrations/edit.html.erb [hunk 1]
```diff
@@ -14,15 +14,43 @@
...
+    <%= f.radio_button :status, :reviewer,
+                       checked: resource.status == 'reviewer' %>
+    <%= f.label :status_reviewer, "reviewer" %>
+    <%= f.radio_button :status, :editor,
+                       checked: resource.status == 'editor' %>
+    <%= f.label :status_editor, "editor" %>
+    <%= f.radio_button :status, :manager,
+                       checked: resource.status == 'manager' %>
+    <%= f.label :status_manager, "manager" %>
+  </fieldset>
```
Comment (AudreyAltman) [inline app/views/registrations/edit.html.erb]: Could we add some brief text explaining the difference permissions afforded to reviewer, editor, and manager?

**Comment by markbreedlove** [comment]
@AudreyAltman: I've added some explanatory text that might clarify the roles.

**Comment by AudreyAltman** [comment]
When testing this in my local browser, I'm finding that I can't change an admin's role/status.    If I am logged in as a manager, access an `admins#registrations#edit` form for myself or any other admin, and try to change the role, it doesn't persist.

**Comment by markbreedlove** [comment]
OK, I'll look into that!

**Comment by markbreedlove** [comment]
@AudreyAltman that last commit oughta fix it.

**Comment by AudreyAltman** [comment]
@markbreedlove yep, that fixed it!  

I think it would be good on the source_set, set, and guide views to hide links to functionality that isn't available to reviewers, but that seems out-of-scope for this ticket - I will make a new Redmine ticket to follow up on it.

All in all, this looks great.  I think it's ready to go.  Do you want to :squash: before I :merge:?

**Comment by markbreedlove** [comment]
Oh yeah, I guess I overlooked some places where there are create-content links. Do the main navigational links at least get removed for you if you're a reviewer? (The links to "Administrators" and so forth?)

**Comment by markbreedlove** [comment]
I will squash it if all's well with the nav links.

**Comment by AudreyAltman** [comment]
@markbreedlove yes, the navigational links accurately reflect what actions are available to all three roles.

**Comment by markbreedlove** [comment]
OK, @AudreyAltman, should be good to go.

**Comment by AudreyAltman** [comment]
:rocket:

**Pull request merged**